client/lib/i18n-utils/utils.js

```diff
@@ -182,23 +182,44 @@ const setLocalizedUrlHost = ( hostname, validLocales = [] ) => ( urlParts, local
 	return urlParts;
 };
 
-const setLocalizedWpComPath = ( prefix, validLocales = [] ) => ( urlParts, localeSlug ) => {
+const setLocalizedWpComPath = ( prefix, validLocales = [], limitPathMatch = null ) => (
+	urlParts,
+	localeSlug
+) => {
 	urlParts.host = 'wordpress.com';
+	if (
+		typeof limitPathMatch === 'object' &&
+		limitPathMatch instanceof RegExp &&
+		! urlParts.pathname.match( limitPathMatch )
```

Comment:
Nitpick: I'd use `limitPathMatch.test( urlParts.pathname )` as it's a better fit in this use case (also has better performance, even though it probably won't be noticeable here).

Reply:
Thanks! Changed it.